### ISSUE TITLE
no member named 'IsSystemVSharedMemory' in sandbox::SyscallSets

### ISSUE THREAD

**Issue by jessesimpson36** [shawnanastasio/chromium_power#2]
```
../../services/service_manager/sandbox/linux/bpf_gpu_policy_linux.cc:69:24: error: no member named 'IsSystemVSharedMemory' in 'sandbox::SyscallSets'
      if (SyscallSets::IsSystemVSharedMemory(sysno))
          ~~~~~~~~~~~~~^
```

It appears that there is an undefined function named IsSystemVSharedMemory(int).

using some basic searching tools, the above file is referencing the following files:
```
  services/service_manager/sandbox/linux/bpf_gpu_policy_linux.cc
  sandbox/linux/seccomp-bpf-helpers/syscall_sets.h
  sandbox/linux/seccomp-bpf-helpers/syscall_sets.cc
```

Since I don't know what the implications are of this function, I intend to try compiling it with and without adding additional logic for power systems on line 69 of bpf_gpu_policy_linux.cc.

I'll report here with what I find, but if you have any insight on that function, I would love to hear it.

**Comment by jessesimpson36** [comment]
```
diff --git a/services/service_manager/sandbox/linux/bpf_gpu_policy_linux.cc b/services/service_manager/sandbox/linux/bpf_gpu_policy_linux.cc
index 1bf2c8c30f..a4da0b5b99 100644
--- a/services/service_manager/sandbox/linux/bpf_gpu_policy_linux.cc
+++ b/services/service_manager/sandbox/linux/bpf_gpu_policy_linux.cc
@@ -65,7 +65,7 @@ ResultExpr GpuProcessPolicy::EvaluateSyscall(int sysno) const {
       if (SyscallSets::IsEventFd(sysno))
         return Allow();

-#if defined(OS_LINUX) && !defined(OS_CHROMEOS) && defined(USE_X11)
+#if defined(OS_LINUX) && !defined(OS_CHROMEOS) && defined(USE_X11) && !defined(__powerpc64__)
       if (SyscallSets::IsSystemVSharedMemory(sysno))
         return Allow();
 #endif
```
I found that I get further in the compilation process by applying the above patch.

**Comment by shawnanastasio** [comment]
ifdef'ing out the call is incorrect. That function is responsible for determining whether a given system call number is used for SysV shm and influences the generated sandbox policy. Without that check there, anything that runs under the sandbox will fail if it tries to use SysV shm.

One of the patches should enable IsSystemVSharedMemory for ppc64, but it was likely broken by an upstream refactor. I'll rebase all the patches on the latest Chromium commit and let you know when it's fixed.

**Comment by shawnanastasio** [comment]
Fixed in e6b95c7da286eb76b21ed59556822a633e4837e9
